### PR TITLE
Fix encode & decode values for approval calls

### LINKED ISSUE #221: 'Vesting contract for DXD token' action - Unknown token
**Description**
'Unknown token' appears even if I have selected asset token while creating Proposal

**Steps to Reproduce**
1. SpinUp Localhost env
2. Open localhost:3000
3. Check if Landing Page is loaded
4. Connect wallet to Metamask and join SWPRGuild
5. Click on 'Create Proposal' button
6. Add 'Vesting contract for DXD token' action with SWPR asset
7. Finish creating Proposal
8. Check Proposal on SWPRGuild page
9. Check Proposal page

**Expected behavior**
Should see name of the selected asset

**Screenshots**
vesting1
vesting2

**Device Information**
OS: MacOs
Browser: GoogleChrome, Brave
Device: MacOS Monterey v12.4

**Additional information**
From #198 
"- Added "Unkown token" to approval summary if data not found"

### PR DESCRIPTION
**Pull request by MiltonTulli**
# Description
Fix approval call fields when encoding and decoding 

Closes https://github.com/DXgovernance/DAVI/issues/197
Closes #221 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

# Checklist:

- [ ] My code follows the existing style of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any UI changes have been tested and made responsive for mobile views

## Before: 
Screen Shot 2022-08-18 at 17 21 01

## After: 
Screen Shot 2022-08-18 at 17 20 20

### CODE REVIEW

**Review by Kenny-Gin1 — APPROVED** [review]
LGTM! And also, I respect that you're making great refactors in the code. It gets a little bit clearer each time. Good thing with the comments as well!

File: src/hooks/Guilds/contracts/useEncodedCall.ts [hunk 1]
```diff
@@ -33,13 +33,13 @@ export const bulkEncodeCallsFromOptions = (options: Option[]): Option[] => {
         };
         if (!!decodedAction.approval) {
           const approvalCall = {
-            from: decodedAction.decodedCall.from,
-            to: decodedAction.approval?.token,
+            from: decodedAction.decodedCall.from, // Guild address
+            to: decodedAction.approval?.token, // Token address
             data: encodeAprovalCall(
```
Comment (Kenny-Gin1) [inline src/hooks/Guilds/contracts/useEncodedCall.ts:38]: 🖊️ typo `encodeApprovalCall`

**Comment by Filipv95** [comment]
@MiltonTulli I've check this PR as well and it looks good! @rossneilson I think this closes #221?

**Comment by rossneilson** [comment]
> @MiltonTulli I've check this PR as well and it looks good! @rossneilson I think this closes #221?

Correct it should close this also!

File: src/hooks/Guilds/contracts/useEncodedCall.ts [hunk 2]
```diff
@@ -33,13 +33,13 @@ export const bulkEncodeCallsFromOptions = (options: Option[]): Option[] => {
...
         if (!!decodedAction.approval) {
           const approvalCall = {
-            from: decodedAction.decodedCall.from,
-            to: decodedAction.approval?.token,
+            from: decodedAction.decodedCall.from, // Guild address
+            to: decodedAction.approval?.token, // Token address
             data: encodeAprovalCall(
-              decodedAction.decodedCall.from,
-              decodedAction.approval?.amount
+              decodedAction.decodedCall.to, // Spender: Contract we are doing the actual spending call to
```
Comment (rossneilson) [inline src/hooks/Guilds/contracts/useEncodedCall.ts:39]: I think the spender is still being set incorrectly, looks like it is being set to the guild ID

Screenshot 2022-08-19 at 10 15 48

**Review by rossneilson — CHANGES_REQUESTED** [review]

File: src/hooks/Guilds/contracts/useEncodedCall.ts (hunk 2 above)
Comment (MiltonTulli) [reply src/hooks/Guilds/contracts/useEncodedCall.ts:39]: Weird. Let me test again.

Comment (MiltonTulli) [reply src/hooks/Guilds/contracts/useEncodedCall.ts:39]: Fixed. Was only UI stuff displaying the wrong value, values on the call are ok!

File: src/hooks/Guilds/contracts/useEncodedCall.ts (hunk 1 above)
Comment (MiltonTulli) [reply src/hooks/Guilds/contracts/useEncodedCall.ts]: Sure. Fixed

**Review by rossneilson — APPROVED** [review]

**Pull request merged**